http-common/Cargo.toml

```diff
@@ -30,3 +30,4 @@ serde_json = "1"
 
 [features]
 tokio1 = ["headers", "hyper", "hyper-openssl", "hyper-proxy", "openssl", "openssl-sys", "tokio"]
+serve_with_shutdown = ["tokio/time"]
```

Comment:
Why do we need it to be optional? Might as well convert `aziotd` to use it too? (In which case, keep the old function name `serve`.)

Reply:
Do we need graceful shutdown in aziotd? AFAIK, we don't handle CTRL+C here or have any APIs that would cause any service to restart.

Reply:
Well, it would simplify this http-common code. aziotd's main can just keep the `Sender` around and not use it. Just make sure not to drop it, otherwise the `Receiver` will resolve with `Err(_)`

Reply:
Ok, makes sense. Fixed.